api/proxy/proxy.go

```diff
@@ -320,12 +331,37 @@ func (p *ClientProxy) Start(opt ...Option) (retErr error) {
 					// connections
 					return
 				}
+			default:
```

Comment:
Will this loop not busy loop now? The default cause will be hit almost every time. Are we waiting somewhere else where before we should wait for the context or connsLeftCh to be readable?